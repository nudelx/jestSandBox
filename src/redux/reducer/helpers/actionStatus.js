const actionStatus = {
  FETCHING: 'FETCHING',
  DONE: 'DONE',
  FAIL: 'FAIL'
}

export default actionStatus
