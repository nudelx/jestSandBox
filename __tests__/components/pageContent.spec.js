import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import PageContent from '../../src/components/PageContent'
import toJson from 'enzyme-to-json'
import actionStatus from '../../src/redux/reducer/helpers/actionStatus'

Enzyme.configure({ adapter: new Adapter() })

const movie = {
  page_1: {
    data: {
      title: 'test title',
      opening_crawl: 'this is a crawl',
      director: 'BB-8',
      release_date: '1/1/1970'
    }
  },
  status: 'DONE'
}

const content = {
  currentPage: 'page_1'
}

describe('page Content test ', () => {
  it('empty card should match the snapshot without props (empty)', () => {
    const component = shallow(<PageContent />)
    expect(toJson(component)).toMatchSnapshot()
  })

  it('empty card should match the snapshot with props (<movie>)', () => {
    const component = shallow(<PageContent content={content} movies={movie} />)
    expect(toJson(component)).toMatchSnapshot()
  })

  it('empty card should match the snapshot with loading card (<loading>)', () => {
    movie.page_1.status = actionStatus.FETCHING
    const component = shallow(<PageContent content={content} movies={movie} />)
    expect(toJson(component)).toMatchSnapshot()
  })
})
