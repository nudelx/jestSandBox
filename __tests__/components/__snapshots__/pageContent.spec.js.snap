// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page Content test  empty card should match the snapshot with loading card (<loading>) 1`] = `
<div
  className="page-content"
>
  <LoadingCard />
</div>
`;

exports[`page Content test  empty card should match the snapshot with props (<movie>) 1`] = `
<div
  className="page-content"
>
  <WithTrailerDynamic
    movie={
      Object {
        "data": Object {
          "director": "BB-8",
          "opening_crawl": "this is a crawl",
          "release_date": "1/1/1970",
          "title": "test title",
        },
      }
    }
  >
    <Component />
  </WithTrailerDynamic>
</div>
`;

exports[`page Content test  empty card should match the snapshot without props (empty) 1`] = `
<div
  className="page-content"
>
  <EmptyCard />
</div>
`;
